File: ros_ws/src/my_robot_tutorials/scripts/smartphone.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String


def main(args=None):
    rospy.init_node("smartphone")

    subscriber = rospy.Subscriber("robot_news_radio", String, callback_subscription, queue_size=10)    

    rospy.loginfo("smartphone node has been created")    

    rospy.spin()


def callback_subscription(msg):
    rospy.loginfo(msg.data)    


if __name__ == "__main__":
    main()

File: ros_ws/src/my_robot_tutorials/src/smartphone.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback_subscription(const std_msgs::String& msg)
{
    ROS_INFO("%s", msg.data.c_str());
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "smartphone");
    ros::NodeHandle nh;
    ros::Subscriber subscriber = nh.subscribe("robot_news_radio", 10, callback_subscription);

    ros::spin();
}